Sources/BraintreeCore/Authorization/ConfigurationLoader.swift

```diff
@@ -0,0 +1,45 @@
+
+protocol ConfigurationLoadable {
+    func getConfiguration() async throws -> BTConfiguration
+}
+
+final class ConfigurationLoader: ConfigurationLoadable {
+    
+    public var authorization: ClientAuthorization
+    private var cache = ConfigurationCache.shared
+    private lazy var http: BTHTTP? = BTHTTP(authorization: authorization)
+    
+    init(authorization: ClientAuthorization) {
+        self.authorization = authorization
+    }
+    
+    func getConfiguration() async throws -> BTConfiguration {
+        try await withCheckedThrowingContinuation { continuation in
```

Comment:
Nice! Even disregarding prefetch,  I don't think this addresses the multiple network calls if somehow this function is called concurrently.

Reply:
If we assume most merchants have this integration - BTAPIClient, BTPaymentMethodClient, tokenize in one click, I'm happy to disregard prefetching, makes things a lot easier.